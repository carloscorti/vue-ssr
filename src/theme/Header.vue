<template>
  <nav class="navbar has-shadow">
    <div class="container">
      <a href="/">
        <img :src="imgSrc" alt="Vue SPA" />
      </a>
    </div>
  </nav>
</template>

<script>
export default {
  name: 'Header',
  data() {
    return {
      imgSrc: 'http://bit.ly/vue-img'
    };
  }
};
</script>
