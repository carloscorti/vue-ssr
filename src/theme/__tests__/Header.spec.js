import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Header from '../Header.vue';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);

describe('Header', () => {
  let state;
  let getters;
  let store;
  let router;

  beforeEach(() => {
    state = {
      isAuthenticated: false
    };

    getters = {
      isAuthenticated: state => state.isAuthenticated
    };

    store = new Vuex.Store({
      state,
      getters
    });

    router = new VueRouter();
  });

  it('renders correctly with isAuthenticated false', () => {
    const wrapper = shallowMount(Header, { router, store, localVue });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with isAuthenticated true', async () => {
    const wrapper = shallowMount(Header, { router, store, localVue });
    wrapper.vm.$store.state.isAuthenticated = true;
    await localVue.nextTick();
    expect(wrapper.find('span').text()).toBe('Logout');
  });
});
