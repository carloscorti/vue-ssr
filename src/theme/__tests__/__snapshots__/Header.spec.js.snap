// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with isAuthenticated false 1`] = `
<nav class="navbar has-shadow">
  <div class="container">
    <router-link-stub to="[object Object]" tag="a" ariacurrentvalue="page" event="click"><img src="http://bit.ly/vue-img" alt="Vue SPA"></router-link-stub>
    <router-link-stub to="/category/front-end" tag="a" ariacurrentvalue="page" event="click" class="navbar-item is-tab">Front-End</router-link-stub>
    <router-link-stub to="[object Object]" tag="a" ariacurrentvalue="page" event="click" class="navbar-item is-tab">Movile</router-link-stub>
    <router-link-stub to="/login" tag="a" ariacurrentvalue="page" event="click" class="navbar-item is-tab"><span>Login</span></router-link-stub>
  </div>
</nav>
`;
